mlir/lib/IR/BuiltinTypes.cpp

```diff
@@ -91,6 +91,8 @@ IntegerType IntegerType::scaleElementBitwidth(unsigned scale) {
 //===----------------------------------------------------------------------===//
 
 unsigned FloatType::getWidth() {
+  if (llvm::isa<Float6E3M2FNType>(*this))
+    return 6;
```

Comment:
Why don't we use APFloat here?

Reply:
I'll do this in a separate PR, if you don't mind.

Reply:
Created https://github.com/llvm/llvm-project/pull/107372